ext/strings_test.go

```diff
@@ -1254,36 +1247,38 @@ func TestStringFormat(t *testing.T) {
 		}
 		return opts
 	}
-	buildOpts := func(locale string, variables []cel.EnvOption) []cel.EnvOption {
+	buildOpts := func(skipCompileCheck bool, locale string, variables []cel.EnvOption) []cel.EnvOption {
 		opts := []cel.EnvOption{
-			Strings(StringsLocale(locale)),
+			Strings(StringsLocale(locale), StringsValidateFormatCalls(!skipCompileCheck)),
 			cel.Container("ext"),
 			cel.Abbrevs("google.expr.proto3.test"),
 			cel.Types(&proto3pb.TestAllTypes{}),
-			cel.ASTValidators(cel.ValidateHomogeneousAggregateLiterals()),
 			NativeTypes(
 				reflect.TypeOf(&TestNestedType{}),
 				reflect.ValueOf(&TestAllTypes{}),
 			),
 		}
+		if !skipCompileCheck {
+			opts = append(opts, cel.ASTValidators(cel.ValidateHomogeneousAggregateLiterals()))
```

Comment:
I'm a little confused as to why `ValidateHomogeneousAggregateLiterals` got moved inside this guard here - is there some kind of additional checking that will accidentally get caught if we check literals while not checking `string.format`?

Reply:
I can leave it enabled. I can't recall why I moved it.